pkgs/os-specific/linux/dcgm/default.nix

```diff
@@ -0,0 +1,113 @@
+{ lib
+, callPackage
+, gcc11Stdenv
+, fetchFromGitHub
+, catch2
+, cmake
+, cudaPackages_10_2
+, cudaPackages_11_8
+, cudaPackages_12
+, fmt
+, git
+, jsoncpp
+, libevent
+, libyamlcpp
+, plog
+, python3
+, tclap_1_4
+
+# Allow builds targeting non-NixOS distros to adjust where libcuda.so is found
+# since it should not be pinned by Nix.
+, addOpenGLRunpath
+, nvidiaDriverPaths ? [ "${addOpenGLRunpath.driverLink}/lib" ]
+}:
+let
+  # Flags copied from DCGM's libevent build script
+  libevent-nossl = libevent.override { sslSupport = false; };
+  libevent-nossl-static = libevent-nossl.overrideAttrs (super: {
+    CFLAGS = "-Wno-cast-function-type -Wno-implicit-fallthrough -fPIC";
+    CXXFLAGS = "-Wno-cast-function-type -Wno-implicit-fallthrough -fPIC";
+    configureFlags = super.configureFlags ++ [ "--disable-shared" "--with-pic" ];
+  });
+
+  jsoncpp-static = jsoncpp.override { enableStatic = true; };
+
+  cudaToolkits = [
+    {
+      version = "10";
+      toolkit = cudaPackages_10_2.cudatoolkit;
+    }
+    {
+      version = "11";
+      toolkit = cudaPackages_11_8.cudatoolkit;
+    }
+    {
+      version = "12";
+      toolkit = cudaPackages_12.cudatoolkit;
+    }
+  ];
+
+  # Builds CMake code to add CUDA paths for include and lib.
+  mkAppendCudaPaths = { version, toolkit }: ''
+    list(APPEND Cuda${version}_INCLUDE_PATHS "${toolkit}/include")
+    list(APPEND Cuda${version}_LIB_PATHS "${toolkit}/lib" "${toolkit.lib}/lib")
+  '';
+in gcc11Stdenv.mkDerivation rec {
+  pname = "dcgm";
+  version = "3.1.8";
+
+  src = fetchFromGitHub {
+    owner = "NVIDIA";
+    repo = "DCGM";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-OXqXkP2ZUNPzafGIgJ0MKa39xB84keVFFYl+JsHgnks=";
+  };
+
+  # Add our paths to the CUDA paths so FindCuda.cmake can find them.
+  EXTRA_CUDA_PATHS = lib.strings.concatMapStringsSep "\n" mkAppendCudaPaths cudaToolkits;
```

Comment:
It does look like that would be difficult to untangle, I thought it might be possible to substitute at use but `FindCMake` hardcodes the 3 versions to load.

Depending on how these proxies are linked it might be possible to make them use `cudaPackages.libcublas` instead of the full toolkit and cut down on the closure size.

Reply:
I could not find a way to do this.